cs/quickstart/refactoring-model.texy

```diff
@@ -0,0 +1,121 @@
+
+Pokud bychom nevyužívali frameworku a síly DI containeru, museli bychom v našem případě nejprve vytvořit instaci `Nette\Database\Connection`, `Nette\Database\Structure`, `Nette\Database\Conventions` a `Nette\Caching\Storages\FileStorage` (které pro své instancování také potřebují nějaké závislosti, třeba jméno a heslo k databázi) pro to, abychom mohli vytvořit instanci `Nette\Database\Context`, kterou bychom následně použili pro vytvoření instance `ArticleManager`.
+
+/--php
+	...
+	// vyrobime vse co je treba pro Context
+	$dbConnection = new Nette\Database\Connection(...);
+	$dbStructure = new Nette\Database\Structure(...);
+	$dbConventions = new Nette\Database\Conventions\DiscoveredConventions(...);
+	$dbCache = new Nette\Caching\Storages\FileStorage(...);
+
+	// vyrobime Context
+	$dbContext = new Nette\Database\Context($dbConnection, $dbStructure, $dbConventions, $dbCache);
+
+	// vyrobime ArticleManager
+	$articleManager = new App\Model\ArticleManager($dbContext)
+	...
+\--
+
+Tolik psaní pro to, abychom dostali fungujicí ArticleManager? To asi nechceme, proto použijeme DI container, který berme jako skřínku co generuje tyto výrobní kódy a nám vrátí jen požadovaný objekt.
+
+Jediné co potřebujeme, aby nám DI container vrátil námi požadovaný objekt, je mu řící, že ho budeme chtít. To se dělá v konfiguraci aplikace. V souboru `app/config/config.neon` v sekci services přidáme řádek s celým názvem třídy.
+
+/--neon
+services:
+	- App\Forms\SignFormFactory
+	router: App\RouterFactory::createRouter
+	- App\Model\ArticleManager
+\--
+
+
+Tím Nette a jeho "DI containeru":[dependency-injection] říkáme, kdyby někdo chtěl instanci této třídy, tak víš kde ji hledat, dej ji vše co potřebuje k životu a připrav nám ji pro použití v dalších třídách. Takto registrovaná třída se nazývá **služba**.
+
+
+Přepneme se do `HomepagePresenter.php`, který upravíme tak, že se zbavíme závislosti na `Nette\Database\Context` a nahradíme za novou závislost na naší nové třídě.
+
+
+/--php
+<?php
+
+namespace App\Presenters;
+
+use Nette,
+	App\Model\ArticleManager;
+
+/**
+ * Homepage presenter.
+ */
+class HomepagePresenter extends BasePresenter
+{
+	/** @var ArticleManager */
+	private $articleManager;
+
+	public function __construct(ArticleManager $articleManager)
+	{
+		$this->articleManager = $articleManager;
+	}
+
+	public function renderDefault()
+	{
+		$this->template->posts = $this->articleManager->getPublicArticles()->limit(5);
+	}
+
+}
+
+
+\--
+.[tip]
+V sekci use máme `App\Model\ArticleManager`, tak si můžeme zápis v php kódu zkrátit na ArticleManager (nebojte, i v komentářích to funguje a vaše chytré IDE by si s tím mělo poradit).
+
+
+
+V konstrukotu si požádáme o `ArticleManager`, který si přiřadíme do private property a v metodě renderDefault zavoláme metodu `getPublicArticles()` a nad výsledkem, co se nám vrátí, zavoláme ještě metodu `limit(5)`;
```

Comment:
typo: V **konstruktoru**... :-P

Asi bych počeštil "private property". Jak se tomu říká správně v tomto kontextu? Jen proměnná, nebo vlastnost? Nenapadá mě teď to správné slovo... :-)